### PR TITLE
BUG: Copy meshgrid after broadcasting

### LINKED ISSUE #8561: Assigning values not working properly on arrays returned by meshgrid
On the current master branch, trying to assign a value to certain parts of an array returned by `meshgrid` isn't working properly:
```
import numpy as np
X = np.array([1.1, 2.2])
Y = np.array([3.3, 4.4])
x, y = np.meshgrid(X, Y)

print(x)
print(x[0, :])
x[0, :] = 0
print(x)
```
This prints
```
[[ 1.1  2.2]
 [ 1.1  2.2]]

[ 1.1  2.2]

[[ 0.  0.]
 [ 0.  0.]]
```
In this example only the top row of `x` should be set to zero, but the whole `x` array is set to zero. This is working correctly in version 1.1.12. This was discovered on https://github.com/matplotlib/matplotlib/issues/7686 and apparently was introduced by https://github.com/numpy/numpy/pull/5302

### PR DESCRIPTION
**Pull request by eric-wieser**
Also, remove some unused variables

Fixes #8561

### CODE REVIEW

File: numpy/lib/stride_tricks.py [hunk 1]
```diff
@@ -179,7 +179,7 @@ def _broadcast_shape(*args):
     supplied arrays against each other.
     """
     if not args:
-        raise ValueError('must provide at least one argument')
+        return ()
     # use the old-iterator because np.nditer does not handle size 0 arrays
     # consistently
     b = np.broadcast(*args[:32])
```
Comment (eric-wieser) [inline numpy/lib/stride_tricks.py]: Is there any reason that `np.broadcast()` is forbidden?

Comment (mhvk) [inline numpy/lib/stride_tricks.py]: Agreed.

**Comment by mhvk** [comment]
This look all OK to me; missing only is a possible addition to the release notes (this one is probably a bit more important to add at all than #8618, which arguably is not something that people have to be specifically alerted to).

**Comment by eric-wieser** [comment]
AFAICT, this regression was introduced after 1.12, so doesn't need to make it into 1.12.1?

**Comment by mhvk** [comment]
Yes, you're right! So, I think we're set then, and I'll use my newly acquired powers...

**Comment by eric-wieser** [comment]
@mhvk: I'm curious, what's the upgrade path for acquiring some of those powers? Being able to stick labels on my PRs would be handy

File: numpy/lib/stride_tricks.py (hunk 1 above)
Comment (eric-wieser) [inline numpy/lib/stride_tricks.py]: @mhvk: So you'd be in favor of a separate PR that removes that restriction from the C code?

**Comment by mhvk** [comment]
I just was asked a few days ago to join the team, so now I can set labels and merge...  I do wonder whether it would not make sense to let everyone at least set labels on their own PRs.

File: numpy/lib/stride_tricks.py (hunk 1 above)
Comment (mhvk) [inline numpy/lib/stride_tricks.py]: I slightly misunderstood -- I agreed with your change here for sure. I'm less sure about messing with the C code, but this is mostly because quite a while ago, on earlier changes in `stride_tricks`, there was the following https://github.com/numpy/numpy/pull/5371#issuecomment-67233157

**Comment by rgommers** [comment]
> I do wonder whether it would not make sense to let everyone at least set labels on their own PRs.

That's not possible given the not very granular permissions available:
https://help.github.com/articles/repository-permission-levels-for-an-organization/
https://github.com/isaacs/github/issues/311

> I'm curious, what's the upgrade path for acquiring some of those powers? Being able to stick labels on my PRs would be handy

Usually people don't ask but just continue sending and reviewing good PRs until a core dev finally wakes up and approaches them. However, people *are* very welcome to ask or express the ambition to join the core team (either in private to a core dev or publicly). And since you did just ask: your high activity level hasn't gone unnoticed @eric-wieser! I'll bring it up with the rest of the team and get back to you.

**Comment by rgommers** [comment]
I tried to find a description of how we manage commit rights works in https://docs.scipy.org/doc/numpy-dev/dev/, either in the contributor or governance section, but it's not explained. Will have a look at fixing that.
